### PR TITLE
Ci update

### PR DESCRIPTION
**Pull request by HubertBalcerzak**

### CODE REVIEW

**Comment by loumalouomega** [comment]
Are we officially on C++14?, @RiccardoRossi said in a PR something of using boost because of C++11 and now I am confused.

File: scripts/build/gitlab/configure_gitlab_windows.bat [hunk 1]
```diff
@@ -16,7 +16,7 @@ cd ..\..\..\cmake_build
 -DEXTERNAL_SOLVERS_APPLICATION=OFF                                              ^
 -DFLUID_DYNAMICS_APPLICATION=ON                                                 ^
 -DSTRUCTURAL_MECHANICS_APPLICATION=ON                                           ^
--DCONTACT_STRUCTURAL_MECHANICS_APPLICATION=ON                                   ^
+-DCONTACT_STRUCTURAL_MECHANICS_APPLICATION=OFF                                  ^
```
Comment (loumalouomega) [inline scripts/build/gitlab/configure_gitlab_windows.bat]: Why?, you need to ask first

Comment (loumalouomega) [inline scripts/build/gitlab/configure_gitlab_windows.bat]:
```suggestion
-DCONTACT_STRUCTURAL_MECHANICS_APPLICATION=ON                                  ^
```

**Review by loumalouomega — COMMENTED** [review]
Nobody told me my application would be removed from CI

File: scripts/build/gitlab/configure_gitlab_windows.bat (hunk 1 above)
Comment (philbucher) [reply scripts/build/gitlab/configure_gitlab_windows.bat]: @loumalouomega this is the new CI, not appveyor
currently @Hubertus248 is testing it, and it is not working hence he tries to lower the requirements

I hope I am not wrong @Hubertus248

Comment (loumalouomega) [reply scripts/build/gitlab/configure_gitlab_windows.bat]: Ok, I have not blocked, just commented

**Comment by roigcarlo** [comment]
@loumalouomega the C++14 was a suggestion of mine to try to solve an error, but I think we can leave it.

**Comment by loumalouomega** [comment]
> @loumalouomega the C++14 was a suggestion of mine to try to solve an error, but I think we can leave it.

Just asking, current standard is C++14. In fact according to Wikipedia C++17 is the current standard https://es.wikipedia.org/wiki/C%2B%2B

**Comment by HubertBalcerzak** [comment]
@loumalouomega This application won't compile on Windows, because it's name is too long and it violates Windows path length limit during compilation. Yes, I also love this system.

https://gitlab.com/KratosMultiphysics/Kratos/-/jobs/238601322 Error at the very end of this log.

File: scripts/build/gitlab/configure_gitlab_windows.bat (hunk 1 above)
Comment (HubertBalcerzak) [reply scripts/build/gitlab/configure_gitlab_windows.bat]: After more tinkering build finishes with application enabled, so it's not required anymore

**Review by roigcarlo — APPROVED** [review]
Approving, we will need to change more things but the scope of the issue is solved.

**Comment by philbucher** [comment]
is the branch still needed?

**Comment by pooyan-dadvand** [comment]
Still I didn't understood why you had to move to c++14? Considering that this would fail on gcc 4.8 (or oldest supported compiler)

**Comment by roigcarlo** [comment]
@pooyan-dadvand We can turn that back, It was just to check that some of the early problems.
We can restore the c++11 back when we fix the current errors with the tests.

We are not using it in Linux, but all implementations of MSVC that have C+11 are implementing C++14 as well, and there is no way to my knowledge to tell the Microsoft compiler to enforce 11 standard over 14, the tests I've done keep working on syntax that should fail. Still as I said, we will remove the flag in the next update.

**Comment by pooyan-dadvand** [comment]
but you are changing the trusty.sh Isn't it?

**Comment by roigcarlo** [comment]
Ouch didn't see that. We will change it asap

**Pull request merged**